src/utils/wallet.js

```diff
@@ -740,12 +755,23 @@ class Wallet {
         await Promise.all(accountIds.map(async (accountId, i) => {
             if (!accountId || !accountId.length) return
 
-            // set up temp wallet instance
+            // temp account
             this.connection = connection
             this.accountId = accountId
             this.twoFactor = new TwoFactor(this)
             this.twoFactor.accountId = accountId
-            const account = this.getAccount(accountId)
+            const has2fa = this.has2fa = await this.twoFactor.isEnabled()
```

Comment:
I fail to understand how this can even work in the first place since you seem to end up modifying a single Wallet instance asyncronously and this should introduce races.

Reply:
removed, unused